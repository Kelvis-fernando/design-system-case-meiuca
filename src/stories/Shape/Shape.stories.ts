import type { Meta, StoryObj } from "@storybook/react";

import { Shape } from "./Shape";

const meta = {
  title: "Components/Shape",
  component: Shape,
  tags: ["autodocs"],
} satisfies Meta<typeof Shape>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: `Ola`,
    width: "100px",
    height: "100px",
  },
};
