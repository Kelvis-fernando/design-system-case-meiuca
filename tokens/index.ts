import colors from "./color.json";
import fonts from "./font.json";
import borders from "./border.json";
import shadows from "./shadow.json";
import spacings from "./spacing.json";

export const tokens = {
  colors,
  fonts,
  borders,
  shadows,
  spacings,
};
