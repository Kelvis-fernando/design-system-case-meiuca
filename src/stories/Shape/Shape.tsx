import { shapeStyle } from "./Shape.style";
import { ShapeProps } from "../../types/ShapeProps";

export const Shape: React.FC<ShapeProps> = ({
  children,
  width,
  height,
  ...props
}) => {
  return (
    <div style={shapeStyle(width!, height!)} {...props}>
      {children}
    </div>
  );
};
