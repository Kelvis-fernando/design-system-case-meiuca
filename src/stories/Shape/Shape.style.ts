import { tokens } from "../../../tokens/index";

export const getShapeStyle = (
  width: number,
  height: number
): React.CSSProperties => {
  return {
    backgroundColor: tokens.colors.neutral.color[1].value,
    boxShadow: `${tokens.shadows.shadow.level[1].stack.value} ${tokens.colors.neutral.color[3].value}`,
    width: width,
    height: height,
    borderRadius: tokens.borders.radius.size.sm.value,
  };
};
