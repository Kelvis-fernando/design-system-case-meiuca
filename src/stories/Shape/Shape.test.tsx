import React from "react";
import { render, screen } from "@testing-library/react";
import { Shape } from "./Shape";
import { tokens } from "../../../tokens";

describe("Shape", () => {
  test("applies the correct style to the shape", () => {
    const width = 200;
    const height = 200;
    const { container } = render(
      <Shape width={width} height={height} data-testid="shape">
        Content
      </Shape>
    );
    const shapeElement = screen.getByTestId("shape");

    expect(container).toMatchSnapshot();
    expect(shapeElement).toHaveStyle(`
    background-color: ${tokens.colors.neutral.color[1].value};
    border-radius: ${tokens.borders.radius.size.sm.value};
    box-shadow: ${tokens.shadows.shadow.level[1].stack.value} ${tokens.colors.neutral.color[3].value};
    height: ${height}px;
    width: ${width}px;
  `);
  });
});
