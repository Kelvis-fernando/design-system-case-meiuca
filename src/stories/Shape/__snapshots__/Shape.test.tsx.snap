// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape applies the correct style to the shape 1`] = `
<div>
  <div
    data-testid="shape"
    style="background-color: rgb(255, 255, 255); box-shadow: 0px 2px 4px #A6AAAF; width: 200px; height: 200px; border-radius: 8px;"
  >
    Content
  </div>
</div>
`;
